packages/ripple-nuxt-tide/lib/core/tide.js

```diff
@@ -108,57 +108,60 @@ export const tide = (axios, site, config) => ({
       'field_site_footer_logos.field_paragraph_media',
       'field_site_footer_logos.field_paragraph_media.field_media_image'
     ]
+
+    if (this.isModuleEnabled('alert')) {
+      include.push(['site_alerts', 'site_alerts.field_alert_type'])
+    }
+
     const menuFields = this.getMenuFields()
     for (let menu in menuFields) {
       include.push(menuFields[menu])
     }
-
     const params = { include: include.toString() }
 
-    let sitesData = await this.getSitesData(params)
+    // let sitesData = null
+    let siteData = null
 
-    if (sitesData instanceof Error) {
-      return sitesData
+    if (siteId !== null) {
+      params.filter = {
+        drupal_internal__tid: {
+          path: 'drupal_internal__tid',
+          value: siteId
+        }
+      }
+      const response = await this.get(`taxonomy_term/sites`, params)
```

Comment:
This will get `undefined` if API request failed. I need to change this to handle error on each request instead of handle it globally in https://github.com/dpc-sdp/ripple/blob/develop/packages/ripple-nuxt-tide/lib/templates/plugin.js#L61. As the error been resolved there to prevent break the site.

What we need to do now, is check if it's `undefined` before pass into `jsonapiParse.parse`.

Reply:
Also, any reason we didn't use `getSitesData()`? We need to handle the pagination as well.

Reply:
as discussed, getSiteData was fetching ALL sites and then just picking the needed one - have changed this to only fetching the needed one using filter param